core/src/test/java/org/apache/iceberg/TestTableUpdatePartitionSpec.java

```diff
@@ -140,6 +140,19 @@ public void testRenameField() {
 
     Assert.assertEquals("should match evolved spec", evolvedSpec, table.spec());
     Assert.assertEquals(1002, table.spec().lastAssignedFieldId());
+
+    table.updateSpec()
+        .renameField("id_bucket_8", "id_partition")
+        .commit();
+
+    evolvedSpec = PartitionSpec.builderFor(table.schema())
+        .withSpecId(3)
+        .bucket("data", 16, "data_bucket")
+        .bucket("id", 8, "id_partition")
+        .truncate("id", 4, "id_trunc_4")
+        .build();
+    Assert.assertEquals("should match evolved spec", evolvedSpec, table.spec());
+    Assert.assertEquals(1002, table.spec().lastAssignedFieldId());
```

Comment:
This seems like it should be its own test case. Could you move it?

Reply:
Updated. Thanks.